auth-svc/middlewares/validation-middleware.js

```diff
@@ -0,0 +1,10 @@
+const { validationResult } = require('express-validator');
+const ApiError = require('../exeptions/api-error');
+
+module.exports = function (req, res, next) {
+  const errors = validationResult(req);
```

Comment:
what does validationResult do? is it validating a specific schema? i'm a little confused here. is this going to be applied only to specific routes?

Reply:
It's extracting validation errors from the request.
So in this middleware, it checks if there are any validation errors, and if so, it returns what is wrong.
In a route '/registration', I use Express Validator as well, so it checks the conditions that the email is in the correct format and the length of the password.
If the requirements are not completed, it will pass them on to the request, and this middleware checks what was incorrect, throws an error, and prevents the user from signing up with incorrect credentials.
Right now, I use it only on the registration route.